docs/source/specs/openapi.yml

```diff
@@ -635,79 +829,124 @@ components:
+            release: 17.5.0
+            version: Darwin Kernel Version 17.5.0
+            machine: x86_64
+            processor: i386
         python_version:
           type: string
-          example: "3.6.1"
+          example: 3.6.1
         modules:
           type: object
-          example: {
-            "coverage": "4.5.1",
-            "coverage.version": "4.5.1",
-            "coverage.xmlreport": "4.5.1",
-            "cryptography": "2.0.3",
-            "ctypes": "1.1.0",
-            "ctypes.macholib": "1.0",
-            "decimal": "1.70",
-            "django": "1.11.5",
-            "django.utils.six": "1.10.0",
-            "django_filters": "1.0.4",
-            "http.server": "0.6",
-          }
+          example:
+            coverage: 4.5.1
+            coverage.version: 4.5.1
+            coverage.xmlreport: 4.5.1
+            cryptography: 2.0.3
+            ctypes: 1.1.0
+            ctypes.macholib: '1.0'
+            decimal: '1.70'
+            django: 1.11.5
+            django.utils.six: 1.10.0
+            django_filters: 1.0.4
+            http.server: '0.6'
         server_id:
           type: string
-          example: "ff4eb8e4-ddfb-4b66-8e0e-045a244990f3"
+          example: ff4eb8e4-ddfb-4b66-8e0e-045a244990f3
     User:
       required:
         - username
         - email
       properties:
         username:
           type: string
-          example: "smithj"
+          example: smithj
         email:
           type: string
           format: email
-          example: "smithj@mytechco.com"
+          example: smithj@mytechco.com
+    UserPreference:
+      required:
+        - name
+        - preference
+      properties:
+        name:
+          type: string
+          example: my-preference-name
+        preference:
+          type: object
+          example: "{'timezone': 'UTC'}"
+        description:
+          type: string
+          example: my long preference description
     UserIn:
       allOf:
-        - $ref: "#/components/schemas/User"
+        - $ref: '#/components/schemas/User'
         - required:
-          - password
+            - password
       properties:
         password:
           type: string
-          example: "str0ng!P@ss"
+          example: str0ng!P@ss
     UserPagination:
       allOf:
-        - $ref: "#/components/schemas/ListPagination"
+        - $ref: '#/components/schemas/ListPagination'
         - required:
-          - results
+            - results
       properties:
         results:
           type: array
           items:
-              $ref: "#/components/schemas/UserOut"
+            $ref: '#/components/schemas/UserOut'
     UserOut:
       allOf:
-        - $ref: "#/components/schemas/User"
+        - $ref: '#/components/schemas/User'
+        - required:
+            - uuid
+      properties:
+        uuid:
+          type: string
+          format: uuid
+          example: 57e60f90-8c0c-4bd1-87a0-2143759aae1c
+    UserPreferenceIn:
```

Comment:
I don't think we need both `UserPreference` and `UserPreferenceIn`.

Your definition adds two `preference` fields when I look at it in the editor:
<img width="494" alt="image" src="https://user-images.githubusercontent.com/29379759/40493568-fdde6336-5f40-11e8-9012-7e094403e367.png">

Reply:
I see both a User and UserIn, Provider and ProviderIn, Customer and CustomerIn.  I'm not sure I understand why Preference would be different.